components/gif_picker/components/App/index.jsx

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+
+import {saveAppProps} from 'mattermost-redux/actions/gifs';
+
+import Header from 'components/gif_picker/components/Header';
+
+const mapDispatchToProps = ({
+    saveAppProps,
+});
+
+export class App extends Component {
```

Comment:
Can you change all the new components to be PureComponent so that they don't re-render as often?